controllers/cloud.redhat.com/clowdjobinvocation_controller.go

```diff
@@ -173,102 +212,244 @@ func (r *ClowdJobInvocationReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
+		VolumeMounts:    []core.VolumeMount{},
 		ImagePullPolicy: core.PullIfNotPresent,
 	}
 
-	if (core.Probe{}) != livenessProbe {
-		c.LivenessProbe = &livenessProbe
-	}
-	if (core.Probe{}) != readinessProbe {
-		c.ReadinessProbe = &readinessProbe
-	}
+	j.Spec.Template.Spec.Volumes = []core.Volume{}
+	configAccess := env.Spec.Providers.Iqe.ConfigAccess
 
-	c.VolumeMounts = append(c.VolumeMounts, core.VolumeMount{
-		Name:      "config-secret",
-		MountPath: "/cdapp/",
-	})
+	switch configAccess {
+	// Build cdenvconfig.json and mount it
+	case "environment":
+		err := r.createAndApplyIqeSecret(ctx, cji, env.Name)
+		if err != nil {
+			r.Log.Error(err, "Cannot apply iqe secret")
+		}
+		c.VolumeMounts = append(c.VolumeMounts, core.VolumeMount{
+			Name:      "cdenvconfig",
+			MountPath: "/cdenv",
+		})
+
+		j.Spec.Template.Spec.Volumes = append(j.Spec.Template.Spec.Volumes, core.Volume{
+			Name: "cdenvconfig",
+			VolumeSource: core.VolumeSource{
+				Secret: &core.SecretVolumeSource{
+					SecretName: env.Name,
+				},
+			},
+		})
+		// if we have env access, we also want app access as well, so also run the next case
+		fallthrough
+
+	// mount cdappconfig
+	case "app":
+		c.VolumeMounts = append(c.VolumeMounts, core.VolumeMount{
+			Name:      "config-secret",
+			MountPath: "/cdapp",
+		})
+
+		j.Spec.Template.Spec.Volumes = append(j.Spec.Template.Spec.Volumes, core.Volume{
+			Name: "config-secret",
+			VolumeSource: core.VolumeSource{
+				Secret: &core.SecretVolumeSource{
+					SecretName: cji.Spec.AppName,
+				},
+			},
+		})
+
+	case "none":
+		r.Log.Info("No config mounted to the iqe pod")
+	}
+	// c.VolumeMounts = append(c.VolumeMounts, core.VolumeMount{
+	// 	Name:      "iqe-settings",
+	// 	MountPath: "/iqe_settings",
+	// })
+
+	// j.Spec.Template.Spec.Volumes = append(j.Spec.Template.Spec.Volumes, core.Volume{
+	// 	Name: "iqe-settings",
+	// 	VolumeSource: core.VolumeSource{
+	// 		Secret: &core.SecretVolumeSource{
+	// 			SecretName: "iqe-settings",
+	// 		},
+	// 	},
+	// })
 
 	j.Spec.Template.Spec.Containers = []core.Container{c}
 
-	j.Spec.Template.Spec.InitContainers = deployProvider.ProcessInitContainers(nn, &c, pod.InitContainers)
+}
 
-	j.Spec.Template.Spec.Volumes = pod.Volumes
-	j.Spec.Template.Spec.Volumes = append(j.Spec.Template.Spec.Volumes, core.Volume{
-		Name: "config-secret",
-		VolumeSource: core.VolumeSource{
-			Secret: &core.SecretVolumeSource{
-				SecretName: cji.Spec.AppName,
-			},
-		},
-	})
+func constructIqeCommand(cji *crd.ClowdJobInvocation, plugin string) []string {
+	command := []string{}
+	command = append(command, "iqe")
+	command = append(command, "tests")
+	command = append(command, "plugin")
+	command = append(command, fmt.Sprintf("%v", strings.ReplaceAll(plugin, "-", "_")))
+	command = append(command, "-m")
+	command = append(command, fmt.Sprintf("%v", cji.Spec.Iqe.Marker))
+	command = append(command, "-k")
+	command = append(command, fmt.Sprintf("%v", cji.Spec.Iqe.Filter))
```

Comment:
I'd prefer to see something like

```
	command := []string{
		"iqe", "tests", "plugin",
		fmt.Sprintf("%v", strings.ReplaceAll(plugin, "-", "_")),
		"-m", fmt.Sprintf("%s", "marker"),
		"-k", fmt.Sprintf("%s", "test-name"),
	}
```

It makes it clearer that the args coming after the `-m` etc are related. Also, Filter is allowed to be empty, which would break this, I think Marker should be allowed to be empty too.

Reply:
I was going to poll for a cleaner way to do this, so big thanks there.